filters/auth/doc.go

```diff
@@ -1,18 +1,128 @@
 /*
-Package auth implements the basic auth for headers based on "https://github.com/abbot/go-http-auth".
+Package auth provides authentication related filters.
 
-How It Works
+Basic - Check Basic Authentication
 
-The filter accepts two parameters, the first mandatory one is the path to the htpasswd file usually used with Apache or nginx. The second one is the optional realm name that will be displayed in the browser.
-Each incoming request will be validated against the password file, for more information which formats are currently
-supported check "https://github.com/abbot/go-http-auth".
-Assuming you are going to use the MD5 version new entries can be generated like
+The filter accepts two parameters, the first mandatory one is the path to the htpasswd file usually used with
+Apache or nginx. The second one is the optional realm name that will be displayed in the browser. Each incoming
+request will be validated against the password file, for more information which formats are currently supported
+check "https://github.com/abbot/go-http-auth". Assuming that the MD5 version will be used, new entries can be generated like
 
 	htpasswd -nbm myName myPassword
 
-Usage
+Embedding the filter in routes:
+
+	generic: * -> basicAuth("/path/to/htpasswd") -> "https://internal.example.org";
+
+	myRealm: Host("my.example.org")
+	  -> basicAuth("/path/to/htpasswd", "My Website")
+	  -> "https://my-internal.example.org";
+
+Set Basic - Outgoing Authentication Header
+
+OAuth - Check Bearer Tokens
+
+The auth filter takes the incoming request, and tries to extract the
+Bearer token from the Authorization header. Then it validates against
+a configured service. Depending on the settings, it also can check if
+the owner of the token belongs to a specific OAuth2 realm, and it can
+check if it has at least one of the predefined scopes. If any of the
+expectations are not met, it doesn't forward the request to the target
+endpoint, but returns with status 401.
+
+As additional features, the package also supports audit logging.
+
+OAuth - Provider Configuration
+
+To enable OAuth2 filters you have to set the CLI argument
+-token-url=<TokenURL>.  Scopes and realms are dependend on your OAuth2
+infrastructure provider. Tokens can be OAuth2 or JWT as long as
```

Comment:
This part: "Tokens can be OAuth2 or JWT as long  as TokenURL returns the data being checked." is not clear to me. Does it mean the format? And that the oauth provider accepts them?

Reply:
The TokenURL service has to accept them, but as @vetinari pointed out this might be a very specific choice to our environment. I try to talk with @vetinari to make this more generic.

Reply:
will be done in the next push